sdk/nodejs/resource.ts

```diff
@@ -486,6 +506,24 @@ export interface CustomTimeouts {
     delete?: string;
 }
 
+/**
+ * ResourceTransformation is the callback signature for the `transformations` resource option.  A
+ * transformation is passed the same set of inputs provided to the `Resource` constructor, and can
+ * optionally return back alternate values for the `props` and/or `opts` prior to the resource
+ * actually being created.  The effect will be as though those props and opts were passed in place
+ * of the original call to the `Resource` constructor.  If the transformation returns undefined,
+ * this indicates that the resource will not be transformed.
+ */
+export type ResourceTransformation =
+    (type: string, name: string, props: Inputs, opts: ResourceOptions) => ResourceTransformationResult | undefined;
+
+/**
+ * ResourceTransformationResult is the result that must be returned by a resource transformation
+ * callback.  It includes new values to use for the `props` and `opts` of the `Resource` in place of
+ * the originally provided values.
+ */
+export type ResourceTransformationResult = { props: Inputs, opts: ResourceOptions };
```

Comment:
1. consider an interface here (though i don't care much).
2. is it never hte case that a transformation would want to tweak a name?  Or the type of something?  Or is that more hte domain of 'aliases'?

Reply:
> consider an interface here (though i don't care much).

Yep - will change.

> is it never hte case that a transformation would want to tweak a name? Or the type of something? Or is that more hte domain of 'aliases'?

Yeah - I tried to come up with cases where that would make sense - but can't come up with anything - and these are a little more challenging as they are more tied to identity.  They are also easy to add in the future - so I thought it would make sense to hold off until we understand better the use cases.